src/pybind/mgr/ssh/module.py

```diff
@@ -149,6 +151,52 @@ def __init__(self, *args, **kwargs):
         # 2. refresh parameter
         self.inventory_cache = orchestrator.OutdatablePersistentDict(self, self._STORE_HOST_PREFIX)
 
+    def _reconfig_ssh(self):
+        temp_files = []
+        ssh_options = []
+
+        # ssh_config
+        ssh_config_fname = self.get_localized_module_option("ssh_config_file")
+        ssh_config = self.get_store("ssh_config")
+        if ssh_config is not None or ssh_config_fname is None:
+            if not ssh_config:
+                ssh_config = DEFAULT_SSH_CONFIG
+            f = tempfile.NamedTemporaryFile(prefix='ceph-mgr-ssh-conf-')
+            os.fchmod(f.fileno(), 0o600);
+            f.write(ssh_config.encode('utf-8'))
+            f.flush() # make visible to other processes
+            temp_files += [f]
+            ssh_config_fname = f.name
+        if ssh_config_fname:
+            if not os.path.isfile(ssh_config_fname):
+                raise Exception("ssh_config \"{}\" does not exist".format(
+                    ssh_config_fname))
+            ssh_options += ['-F', ssh_config_fname]
+
+        # identity
+        ssh_key = self.get_store("ssh_identity_key")
+        ssh_pub = self.get_store("ssh_identity_pub")
+        tpub = None
+        tkey = None
+        if ssh_key and ssh_pub:
+            tkey = tempfile.NamedTemporaryFile(prefix='ceph-mgr-ssh-identity-')
+            tkey.write(ssh_key.encode('utf-8'))
+            os.fchmod(tkey.fileno(), 0o600);
+            tkey.flush() # make visible to other processes
+            tpub = open(tkey.name + '.pub', 'w')
```

Comment:
since this isn't created via tempfile methods, it won't be removed when the process exits

Reply:
Oh yeah, I forgot about this. Any suggestions how to fix it? The TemporaryFile variants don't have a way to specify the name, and I'm not sure how to make python wrapper/finally/destructor thing that cleans it up.

In practice, the mgr python isn't shut down cleanly anyway, so this is kind of a moot point... except when we wire up config changes that might change whether we're usign these files or not. But since it's in the container's private /tmp, it shouldn't really matter anyway.

Reply:
another NamedTemporaryFile with suffix='.pub' should work if the name is a concern. I agree not a big deal though